perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java

```diff
@@ -0,0 +1,369 @@
+
+		final TextBox valueBox = new TextBox();
+		valueBox.getElement().setPropertyString("placeholder", "enter value for new attribute");
+
+		//add attr button
+		CustomButton addButton = TabMenu.getPredefinedButton(ButtonType.ADD, true,
+			ButtonTranslation.INSTANCE.setNewAttributes(), event -> {
+				map = jsonCallback.getMap();
+				if (map.containsValue(keyBox.getText())) {
+					UiElements.generateAlert("", "Key \"" + keyBox.getText() + "\" already exists");
+				} else if (keyBox.getText().equals("")) {
+					UiElements.generateAlert("", "Enter key into key box");
+				} else if (valueBox.isVisible() && !valueBox.getText().equals("")) {
+					addAttribute(keyBox.getText(), valueBox.getText());
+				} else if (!valueBox.isVisible()) {
+					addAttribute(keyBox.getText(), valueBox.getText());
+				} else {
+					UiElements.generateAlert("", "Enter value into value box");
+				}
+			});
+		menu.addWidget(addButton);
+		if (!isAuthorized()) addButton.setEnabled(false);
+
+		// remove attr button
+		final CustomButton removeButton = TabMenu.getPredefinedButton(ButtonType.REMOVE, ButtonTranslation.INSTANCE.removeAttributes());
+		menu.addWidget(removeButton);
+		if (!isAuthorized()) removeButton.setEnabled(false);
+
+
+		// remove button event
+		removeButton.addClickHandler(event -> {
+			ArrayList<Attribute> list = jsonCallback.getTableSelectedList();
+			if (UiElements.cantSaveEmptyListDialogBox(list)) {
+				for (Attribute a : list) {
+					removeAttribute(map.get(a));
+				}
+			}
+		});
+
+		menu.addWidget(keyBox);
+		menu.addWidget(valueBox);
+
+		if (def.getType().equals("java.util.LinkedHashMap") ||
+			def.getType().equals("java.util.ArrayList") ||
+			def.getType().equals("java.util.LargeArrayList") ||
+			def.getType().equals("java.lang.Boolean")) {
+			valueBox.setVisible(false);
+		}
+
+		// add a class to the table and wrap it into scroll panel
+		table.addStyleName("perun-table");
+		ScrollPanel sp = new ScrollPanel(table);
+		sp.addStyleName("perun-tableScrollPanel");
+		if (isAuthorized()) JsonUtils.addTableManagedButton(jsonCallback, table, removeButton);
+
+		// add menu and the table to the main panel
+		firstTabPanel.add(menu);
+		firstTabPanel.setCellHeight(menu, "30px");
+		firstTabPanel.add(sp);
+
+		session.getUiElements().resizePerunTable(sp, 350, this);
+
+		this.contentWidget.setWidget(firstTabPanel);
+
+		return getWidget();
+	}
+
+	@Override
+	public Widget getWidget() {
+		return this.contentWidget;
+	}
+
+	@Override
+	public Widget getTitle() {
+		return this.titleWidget;
+	}
+
+	@Override
+	public ImageResource getIcon() {
+		return SmallIcons.INSTANCE.attributesDisplayIcon();
+	}
+
+	@Override
+	public boolean multipleInstancesEnabled() {
+		return false;
+	}
+
+	@Override
+	public void open() {
+		session.getUiElements().getMenu().openMenu(MainMenu.PERUN_ADMIN);
+		session.getUiElements().getBreadcrumbs().setLocation(0, "edit_keys", getUrlWithParameters());
```

Comment:
This 'edit_keys' should be in a readable form. It is displayed in breadcrumb. I would use something like 'Entityless attributes setting'

Reply:
changed it to sugested 'Entityless attributes setting'